config/routes.rb

```diff
@@ -5,7 +5,6 @@
   root 'home#index'
 
   resources :books
-  # resources :users, only: %i[edit update]
-  resources :users, only: :update
-  get '/settings', to: 'users#edit', as: 'settings'
+  resource :user, only: %i[update edit destroy]
+  resolve('User') { [:user] }
```

Comment:
it should be `resources :users`
Also, why do you use `resolve`?

Reply:
https://guides.rubyonrails.org/routing.html#singular-resources

Reply:
with `resource :user` I have a pretty url and url helper